### DISCUSSION TITLE
IAM concepts

### DISCUSSION THREAD

**Discussion by adriatic** [strapi/strapi#7908] [Legacy Topics]
I would like to initiate the **`IAM Concepts `** Open Source project focusing on the design and implementation of the Strapi interface to a selected set of  Identity and Access Management (IAM) Platform as as Service (PaaS) vendors like Auth0, Okta, AWS IAM, Microsoft MIM, etc. The selection and order of these PaaS vendors have yet to be defined based on Strapi customer needs.

Note that IAM includes support for all identity management tasks that Strapi recently announced (SSO, RBAC, etc) in the industry-standard fashion, with guaranteed security - here is a blob from Okta describing these services:

> For typical cloud based applications (e.g. Salesforce, Google Apps, Workday, etc.), these integrations are delivered as a part of Okta’s Application Network. Administrators simply select from Okta’s list of thousands of supported applications, use a simple wizard answering basic questions about their specific instance of the applications (such as URL and administrative IDs) and Okta handles the rest. All technical details (such as SSO protocols and user management API implementations) are encapsulated in the service and continually maintained by Okta on your behalf. These applications may use a standard like SAML or OpenID, they may use a proprietary API, or they may use Okta’s Secure Web Authentication (SWA) protocol.

> Many of the most popular on-premises web-based applications (Oracle Apps, Lawson, Jira, etc.) are also included in the Okta Application Network. For custom developed on-premises web-based applications Okta provides a range of integration options as well. Secure Web Authentication integration for SSO can be easily added, Okta has SAML toolkits that can be used to SAML enable your apps, and Okta also supports provisioning and de-provisioning into applications that expose user management APIs publicly.

> Okta also provides easy access to mobile enterprise applications from any device. Whether your enterprise apps are HTML5 web apps optimized for mobile platforms or Native iOS or Android apps, Okta has a solution. Any web application in the Okta Application Network can be accessed with a single sign-on from any mobile device. Mobile web apps can use industry-standard SAML, or they can use Okta’s Secure Web Authentication SSO technology. Native applications like Box Mobile can be integrated using SAML authentication for registration and OAuth for ongoing use.

At the moment, there is at least one Strapi user who needs a lot more than Strapi can deliver now. 

Let's discuss the value of this project as well as the interest by Strapi Community to work as project contributors.

**Comment by adriatic** [comment]
Despite zero feedback from the community (?) I had the chance to discuss the meaningfulness of this project, with @derrickmehaffy. So, after realizing that **Strapi uses Passport.js as a Proxy for Oauth2** (as different from Passport.js being a **[provider](https://en.wikipedia.org/wiki/List_of_OAuth_providers)**). This realization changes my plan above, which likely was not correctly understood anyway. I will restate it within days

**Reply by derrickmehaffy** [reply]
Yes that's correct, currently in v3.1.5 we are already using passport (specifically `passport-local`) for the admin panel: 

https://github.com/strapi/strapi/blob/master/packages/strapi-admin/middlewares/auth/index.js

And with the [SSO feature](https://portal.productboard.com/strapi/1-roadmap/c/61-sso-authentication-providers) that will be worked on in Q4 2020, we plan to allow expanding the Admin authentication for other passport methods (Active Directory, SAML, ect) and allowing users to create their own Admin auth providers also using passport.
